go/arrow/array/numeric.gen.go.tmpl

```diff
@@ -82,6 +82,32 @@ func (a *{{.Name}}) setData(data *Data) {
 	}
 }
 
+func (a *{{.Name}}) ValueStr(i int) string {
+	if a.IsNull(i) {
+		return NullValueStr
+	}
+{{if or (eq .Name "Date32") (eq .Name "Date64") -}}
+	return a.values[i].ToTime().Format("2006-01-02")
```

Comment:
these types have a `FormattedString()` method that you can call instead. just do `a.values[i].FormattedString()`